File: in_Java/0796 Rotate String.java
class Solution {
    public boolean rotateString(String A, String B) {
        if(A == null && B == null) return true;
        if(A.length() == 0 && B.length() == 0) return true;
        if(A == null || B == null) return false;
        if(A.length() > B.length()) return false;
        String A2 = A + A;
        for(int i = 0; i < A2.length() - B.length(); i++){
            if(A2.substring(i, i + B.length()).equals(B)) return true;
        }
        return false;
    }
}

File: in_Python/0796 Rotate String.py
class Solution(object):
    def rotateString(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: bool
        """
        if A == None and B == None:
            return True
        if A == None or B == None:
            return False
        if len(A) != len(B):
            return False
        if len(A) == 0 and len(B) == 0:
            return True
        nb = B + B
        #print(nb)
        for i in range(len(nb)):
            j = i + len(A)
            if j > len(nb):
                return False
            if nb[i:j] == A:
                return True
        return False